lib/statsig.ex

```diff
@@ -1,241 +1,72 @@
 defmodule Statsig do
-  use GenServer
-  alias Statsig.Utils
-  require Logger
-
-  # default intervals
-  @flush_interval 60_000
-  @reload_interval 60_000
-
-  def start_link(opts \\ []) do
-    opts =
-      opts
-      |> Keyword.put_new(:name, __MODULE__)
-
-    GenServer.start_link(__MODULE__, opts, name: Keyword.fetch!(opts, :name))
-  end
-
-  def init(opts) do
-    server = Keyword.fetch!(opts, :name)
-
-    :ets.new(ets_name(server), [:named_table])
-
-    Process.flag(:trap_exit, true)
-
-    state = %{
-      api_key: Application.get_env(:statsig, :api_key, nil),
-      last_sync: 0,
-      events: [],
-      tier: Keyword.get(opts, :tier, Application.get_env(:statsig, :env_tier, nil)),
-      prefix: server,
-      flush_interval: Keyword.get(opts, :flush_interval, @flush_interval),
-      reload_interval: Keyword.get(opts, :reload_interval, @reload_interval),
-      reload_timer: nil,
-      flush_timer: nil
-    }
-    {:ok, state}
+  @type options :: %{
+    reload_interval: integer() | nil,
+    flush_interval: integer() | nil
+  }
+  def initialize(options) do
+    reload_interval = Map.get(options, :reload_interval)
+    flush_interval = Map.get(options, :flush_interval)
```

Comment:
nit: It is idiomatic in elixir for options to be Keyword Lists, which are lists of tuples with two elements, the first element in each tuple is an atom. It even has special syntax.
```elixir
```
as opposed to 
```elixir
initialize(reload_interval: :timer.seconds(3), flush_interval: :timer.seconds(5))
```
If you use keyword lists, this becomes:

```suggestion
    reload_interval = Keyword.get(options, :reload_interval)
    flush_interval = Keyword.get(options, :flush_interval)
```

Reply:
I've removed options altogether now, just using application env variables